packages/firebase_remote_config/ios/Classes/FirebaseRemoteConfigPlugin.m

```diff
@@ -0,0 +1,167 @@
+                            1000;
+                        resultDict[@"FETCH_THROTTLED_END"] = [[NSNumber alloc] initWithInt:mills];
+                        NSString *errorMessage =
+                            @"Fetch has been throttled. See the error's FETCH_THROTTLED_END "
+                             "field for throttle end time.";
+                        flutterError = [FlutterError errorWithCode:@"FETCH_FAILED_THROTTLED"
+                                                           message:errorMessage
+                                                           details:resultDict];
+                      } else {
+                        NSString *errorMessage =
+                            @"Unable to complete fetch. Reason is unknown "
+                             "but this could be due to lack of connectivity.";
+                        flutterError = [FlutterError errorWithCode:@"FETCH_FAILED"
+                                                           message:errorMessage
+                                                           details:resultDict];
+                      }
+                      result(flutterError);
+                    } else {
+                      result(resultDict);
+                    }
+                  }];
+  } else if ([@"RemoteConfig#activate" isEqualToString:call.method]) {
+    bool newConfig = [[FIRRemoteConfig remoteConfig] activateFetched];
+    NSDictionary *parameters = [self getConfigParameters];
+    result(@{@"newConfig" : [[NSNumber init] initWithBool:newConfig], @"parameters" : parameters});
+  } else if ([@"RemoteConfig#setDefaults" isEqualToString:call.method]) {
+    FIRRemoteConfig *remoteConfig = [FIRRemoteConfig remoteConfig];
+    NSDictionary *defaults = call.arguments[@"defaults"];
+    [remoteConfig setDefaults:defaults];
+    [[NSUserDefaults standardUserDefaults] setValue:[defaults allKeys] forKey:DEFAULT_KEYS];
+    result(nil);
+  } else {
+    result(FlutterMethodNotImplemented);
+  }
+}
+
+- (NSMutableDictionary *)createRemoteConfigValueDict:(FIRRemoteConfigValue *)remoteConfigValue {
+  NSMutableDictionary *valueDict = [[NSMutableDictionary alloc] init];
+  valueDict[@"value"] = [FlutterStandardTypedData typedDataWithBytes:[remoteConfigValue dataValue]];
+  valueDict[@"source"] =
+      [[NSNumber alloc] initWithInt:[self mapValueSource:[remoteConfigValue source]]];
+  return valueDict;
+}
+
+- (NSDictionary *)getConfigParameters {
+  FIRRemoteConfig *remoteConfig = [FIRRemoteConfig remoteConfig];
+  NSMutableDictionary *parameterDict = [[NSMutableDictionary alloc] init];
+  NSSet *keySet = [remoteConfig keysWithPrefix:@""];
+  for (NSString *key in keySet) {
+    parameterDict[key] = [self createRemoteConfigValueDict:[remoteConfig configValueForKey:key]];
+  }
+  // Add default parameters if missing since `keysWithPrefix` does not return default keys.
+  NSArray *defaultKeys = [[NSUserDefaults standardUserDefaults] arrayForKey:DEFAULT_KEYS];
```

Comment:
You can use the [`allKeysFromSource:`](https://firebase.google.com/docs/reference/ios/firebaseremoteconfig/api/reference/Classes/FIRRemoteConfig#-allkeysfromsourcenamespace) API instead of saving these in `NSUserDefaults`

Reply:
Done